lib/TrackInfo.js

```diff
@@ -238,38 +238,53 @@ class TrackInfo
 
 /**
  * Expands a plain JSON object containing an TrackInfo
- * @param {TrackInfoPlain} plain JSON object
+ * @param {TrackInfoPlain | TrackInfo} plain JSON object
  * @returns {TrackInfo} Parsed Track info
  */
 TrackInfo.expand = function(plain)
 {
+	//Ensure it is not already a TrackInfo object
+	if (plain.constructor.name === "TrackInfo")
+		// @ts-ignore
+		return plain;
+
 	//Create new
 	const trackInfo =  new TrackInfo(plain.media,plain.id);
 	//Check mediaId
 	if (plain.mediaId)
 		//Set it
 		trackInfo.setMediaId(plain.mediaId);
 	//Gor each ssrc
-	for (let i=0;plain.ssrcs && i<plain.ssrcs.length;++i)
+	for (const ssrc of plain.ssrcs || [])
 		//Add ssrc
-		trackInfo.addSSRC(plain.ssrcs[i]);
+		trackInfo.addSSRC(plain.ssrcs[ssrc]);
 	//For each group
-	for (let i=0;plain.groups && i<plain.groups.length;++i)
+	for (const group of plain.groups || [])
 		//Clone and add grou
-		trackInfo.addSourceGroup(SourceGroupInfo.expand(plain.groups[i]));
+		trackInfo.addSourceGroup(SourceGroupInfo.expand(group));
 	//For each encoding
-	for (let i=0;plain.encodings && i<plain.encodings.length;++i)
+	for (const encodings of plain.encodings || [])
 	{
 		const alternatives = [];
 		//For each alternative
-		for (let j=0; j<plain.encodings[i].length;++j)
+		for (const encoding of encodings)
 			//Append it
-			alternatives.push(TrackEncodingInfo.expand(plain.encodings[i][j]));
+			alternatives.push(TrackEncodingInfo.expand(encoding));
 		//Clone and add grou
 		trackInfo.addAlternativeEncodings(alternatives);
 	}
```

Comment:
this code is correct as it is, but it could be better written using map:

~~~ js
for (const alternatives of plain.encodings || [])
	//Clone and add grou
	trackInfo.addAlternativeEncodings(alternatives.map(TrackEncodingInfo.expand));
~~~